diff -Naur a/actor/sharedaction/log_cache_client.go b/actor/sharedaction/log_cache_client.go
--- a/actor/sharedaction/log_cache_client.go	2022-12-14 13:13:23
+++ b/actor/sharedaction/log_cache_client.go	2022-12-14 13:06:18
@@ -5,7 +5,7 @@
 	"time"
 
 	logcache "code.cloudfoundry.org/go-log-cache"
-	"code.cloudfoundry.org/go-loggregator/rpc/loggregator_v2"
+	"code.cloudfoundry.org/go-loggregator/v8/rpc/loggregator_v2"
 )
 
 //go:generate counterfeiter . LogCacheClient
diff -Naur a/actor/sharedaction/logging.go b/actor/sharedaction/logging.go
--- a/actor/sharedaction/logging.go	2022-12-14 13:13:23
+++ b/actor/sharedaction/logging.go	2022-12-14 13:07:03
@@ -10,7 +10,7 @@
 
 	logcache "code.cloudfoundry.org/go-log-cache"
 	"code.cloudfoundry.org/go-log-cache/rpc/logcache_v1"
-	"code.cloudfoundry.org/go-loggregator/rpc/loggregator_v2"
+	"code.cloudfoundry.org/go-loggregator/v8/rpc/loggregator_v2"
 	"github.com/sirupsen/logrus"
 )
 
diff -Naur a/actor/sharedaction/logging_test.go b/actor/sharedaction/logging_test.go
--- a/actor/sharedaction/logging_test.go	2022-12-14 13:13:23
+++ b/actor/sharedaction/logging_test.go	2022-12-14 13:06:23
@@ -10,7 +10,7 @@
 	"code.cloudfoundry.org/cli/actor/sharedaction"
 	"code.cloudfoundry.org/cli/actor/sharedaction/sharedactionfakes"
 	logcache "code.cloudfoundry.org/go-log-cache"
-	"code.cloudfoundry.org/go-loggregator/rpc/loggregator_v2"
+	"code.cloudfoundry.org/go-loggregator/v8/rpc/loggregator_v2"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
diff -Naur a/actor/sharedaction/sharedactionfakes/fake_log_cache_client.go b/actor/sharedaction/sharedactionfakes/fake_log_cache_client.go
--- a/actor/sharedaction/sharedactionfakes/fake_log_cache_client.go	2022-12-14 13:13:23
+++ b/actor/sharedaction/sharedactionfakes/fake_log_cache_client.go	2022-12-14 13:06:27
@@ -8,7 +8,7 @@
 
 	"code.cloudfoundry.org/cli/actor/sharedaction"
 	client "code.cloudfoundry.org/go-log-cache"
-	"code.cloudfoundry.org/go-loggregator/rpc/loggregator_v2"
+	"code.cloudfoundry.org/go-loggregator/v8/rpc/loggregator_v2"
 )
 
 type FakeLogCacheClient struct {
diff -Naur a/actor/v2action/logging_test.go b/actor/v2action/logging_test.go
--- a/actor/v2action/logging_test.go	2022-12-14 13:13:23
+++ b/actor/v2action/logging_test.go	2022-12-14 13:06:13
@@ -13,7 +13,7 @@
 	"code.cloudfoundry.org/cli/api/uaa"
 	"code.cloudfoundry.org/cli/integration/helpers"
 	logcache "code.cloudfoundry.org/go-log-cache"
-	"code.cloudfoundry.org/go-loggregator/rpc/loggregator_v2"
+	"code.cloudfoundry.org/go-loggregator/v8/rpc/loggregator_v2"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
diff -Naur a/actor/v7action/logging_test.go b/actor/v7action/logging_test.go
--- a/actor/v7action/logging_test.go	2022-12-14 13:13:23
+++ b/actor/v7action/logging_test.go	2022-12-14 13:06:03
@@ -12,7 +12,7 @@
 	"code.cloudfoundry.org/cli/api/cloudcontroller/ccv3"
 	"code.cloudfoundry.org/cli/resources"
 	logcache "code.cloudfoundry.org/go-log-cache"
-	"code.cloudfoundry.org/go-loggregator/rpc/loggregator_v2"
+	"code.cloudfoundry.org/go-loggregator/v8/rpc/loggregator_v2"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
