diff --git a/home/files/.vim/vimrc b/home/files/.vim/vimrc
index 3913ae9..be3d748 100644
--- a/home/files/.vim/vimrc
+++ b/home/files/.vim/vimrc
@@ -19,6 +19,9 @@ Plug 'junegunn/fzf.vim'
 Plug 'SirVer/ultisnips', has('python3') ? {} : {'on': []}
 Plug 'honza/vim-snippets'
 
+Plug 'neoclide/coc.nvim', {'branch': 'release'}
+Plug 'junegunn/fzf', {'do': {-> fzf#install()}}
+
 " utilities
 Plug 'direnv/direnv.vim'
 Plug 'tpope/vim-fugitive'
diff --git a/home/files/.zshrc b/home/files/.zshrc
index 9c8c196..367a329 100644
--- a/home/files/.zshrc
+++ b/home/files/.zshrc
@@ -306,5 +306,5 @@ autoload -Uz promptinit && promptinit
 prompt dashboard
 
 # must be run last
-source ~/.nix-profile/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh
+source /usr/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh
 ZSH_HIGHLIGHT_HIGHLIGHTERS+=(brackets)
