diff --git a/home/files/.bashrc b/home/files/.bashrc
index 55aaee6..058e633 100644
--- a/home/files/.bashrc
+++ b/home/files/.bashrc
@@ -42,6 +42,7 @@ GPG_TTY="$(tty)"
 printf -v PATH "%s:" \
   ~/.local/bin \
   "$PATH" \
+  ~/.vim/plugged/fzf/bin \
   "$GOPATH/bin" \
   ~/.config/emacs/bin
 PATH="${PATH%%:}"
diff --git a/home/files/.vim/vimrc b/home/files/.vim/vimrc
index 71ef1f5..72d80bb 100644
--- a/home/files/.vim/vimrc
+++ b/home/files/.vim/vimrc
@@ -19,6 +19,9 @@ Plug 'junegunn/fzf.vim'
 Plug 'SirVer/ultisnips', has('python3') ? {} : {'on': []}
 Plug 'honza/vim-snippets'
 
+Plug 'neoclide/coc.nvim', {'branch': 'release'}
+Plug 'junegunn/fzf', {'do': {-> fzf#install()}}
+
 " utilities
 Plug 'direnv/direnv.vim'
 Plug 'tpope/vim-fugitive'
diff --git a/home/files/.zshrc b/home/files/.zshrc
index f4ed1b8..5520a21 100644
--- a/home/files/.zshrc
+++ b/home/files/.zshrc
@@ -17,6 +17,7 @@ typeset -U path
 path=(
   ~/.local/bin
   $path
+  ~/.vim/plugged/fzf/bin
   "$GOPATH/bin"
   ~/.config/emacs/bin
 )
@@ -310,5 +311,5 @@ autoload -Uz promptinit && promptinit
 prompt dashboard
 
 # must be run last
-source ~/.nix-profile/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh
+source /usr/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh
 ZSH_HIGHLIGHT_HIGHLIGHTERS+=(brackets)
