diff --git a/home/.config/doom/config.el b/home/.config/doom/config.el
index d02dbe5..6e074ac 100644
--- a/home/.config/doom/config.el
+++ b/home/.config/doom/config.el
@@ -13,7 +13,7 @@
 
 
 ;; UI
-(setq doom-font (font-spec :family "Menlo" :size 12)
+(setq doom-font (font-spec :family "Fira Code" :size 12)
       doom-theme 'doom-gruvbox
       display-line-numbers-type t
       frame-title-format t
diff --git a/home/.vim/gvimrc b/home/.vim/gvimrc
index 2c5605f..02be9c1 100644
--- a/home/.vim/gvimrc
+++ b/home/.vim/gvimrc
@@ -9,7 +9,7 @@ set confirm
 if has('gui_macvim')
   set columns=132
   set lines=43
-  set guifont=Menlo\ Regular:h12
+  set guifont=FiraCode-Regular:h12
   set guitablabel=%M%t
   set guioptions-=T
   set transparency=15
