diff --git a/home/.bashrc b/home/.bashrc
index 7783b77..8e00bd9 100644
--- a/home/.bashrc
+++ b/home/.bashrc
@@ -17,7 +17,7 @@ export SYSTEMD_LESS="iFRSMK"
 export GOPATH=~/Documents/src/go
 
 # whether to make use of powerline fonts
-export USE_POWERLINE=0
+export USE_POWERLINE=1
 [[ "$TERM" == "xterm-kitty" ]] && USE_POWERLINE=1
 [[ -z "$DISPLAY$WAYLAND_DISPLAY$SSH_CONNECTION" ]] && USE_POWERLINE=0
 
diff --git a/home/.zshenv b/home/.zshenv
index 3c80a8a..25aca9f 100644
--- a/home/.zshenv
+++ b/home/.zshenv
@@ -7,7 +7,7 @@ export SYSTEMD_LESS="iFRSMK"
 export GOPATH=~/Documents/src/go
 
 # whether to make use of powerline fonts
-export USE_POWERLINE=0
+export USE_POWERLINE=1
 [[ "$TERM" == "xterm-kitty" ]] && USE_POWERLINE=1
 [[ -z "$DISPLAY$WAYLAND_DISPLAY$SSH_CONNECTION" ]] && USE_POWERLINE=0
 
