diff --git a/home/.config/doom/config.org b/home/.config/doom/config.org
index 95a8e8c..636ae3d 100644
--- a/home/.config/doom/config.org
+++ b/home/.config/doom/config.org
@@ -33,7 +33,7 @@ build without having to care much about different displays due to excellent
 autoscaling support out-of-the-box.
 
 #+BEGIN_SRC emacs-lisp
-(setq doom-font (font-spec :family "Menlo" :size 12))
+(setq doom-font (font-spec :family "Fira Code" :size 12))
 #+END_SRC
 ** Frame
 Frame is the term Emacs uses to refer to GUI windows.
diff --git a/home/.config/kitty/kitty.conf b/home/.config/kitty/kitty.conf
index 366dfc0..e180c95 100644
--- a/home/.config/kitty/kitty.conf
+++ b/home/.config/kitty/kitty.conf
@@ -2,16 +2,12 @@
 
 #: Fonts {{{
 
-font_family      Menlo
+font_family      Fira Code
 bold_font        auto
 italic_font      auto
 bold_italic_font auto
 font_size 12.0
 
-# Use the PowerlineSymbols font for rendering Powerline codepoints. Comment
-# this out if your font already supports Powerline glyphs.
-symbol_map U+E0A0-U+E0A3,U+E0C0-U+E0C7 PowerlineSymbols
-
 # Disable multi-character ligatures when the curosr is over them.
 disable_ligatures cursor
 
diff --git a/home/.vim/gvimrc b/home/.vim/gvimrc
index 2c5605f..02be9c1 100644
--- a/home/.vim/gvimrc
+++ b/home/.vim/gvimrc
@@ -9,7 +9,7 @@ set confirm
 if has('gui_macvim')
   set columns=132
   set lines=43
-  set guifont=Menlo\ Regular:h12
+  set guifont=FiraCode-Regular:h12
   set guitablabel=%M%t
   set guioptions-=T
   set transparency=15
