diff --git a/home/.vim/vimrc b/home/.vim/vimrc
index 380dc6c..77e4d22 100644
--- a/home/.vim/vimrc
+++ b/home/.vim/vimrc
@@ -373,8 +373,7 @@ let g:ycm_extra_conf_vim_data=[
   \ '&filetype',
   \ 'g:ycm_python_interpreter_path',
   \ 'g:ycm_python_sys_path']
-
-let g:ycm_python_interpreter_path = ''
+let g:ycm_python_interpreter_path = '~/.local/opt/pyenv/shims/python'
 let g:ycm_python_sys_path = []
 let g:ycm_rust_src_path='~/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src'
 
diff --git a/home/.zshrc b/home/.zshrc
index 48af06d..b49e76d 100644
--- a/home/.zshrc
+++ b/home/.zshrc
@@ -33,6 +33,7 @@ alias la='ls -lAh'
 autoload -Uz zmv
 autoload -Uz br cud fuck
 autoload -Uz fzf-sel fzf-run fzf-loop fzf-gen
+autoload -Uz pyenv rbenv
 
 #################
 #  Directories  #
