diff --git a/home/.vim/vimrc b/home/.vim/vimrc
index a30e6b6..c7e834d 100644
--- a/home/.vim/vimrc
+++ b/home/.vim/vimrc
@@ -373,7 +373,7 @@ let g:ycm_extra_conf_vim_data=[
   \ 'g:ycm_python_interpreter_path',
   \ 'g:ycm_python_sys_path']
 
-let g:ycm_python_interpreter_path = ''
+let g:ycm_python_interpreter_path = '/usr/local/var/pyenv/shims/python'
 let g:ycm_python_sys_path = []
 let g:ycm_rust_src_path='~/.rustup/toolchains/stable-x86_64-apple-darwin/lib/rustlib/src/rust/src'
 
diff --git a/home/.zshrc b/home/.zshrc
index dbab750..07716bb 100644
--- a/home/.zshrc
+++ b/home/.zshrc
@@ -43,6 +43,7 @@ alias sudoedit='sudo -e'
 autoload -Uz zmv
 autoload -Uz br cud fuck
 autoload -Uz fzf-sel fzf-run fzf-loop fzf-gen
+autoload -Uz pyenv rbenv
 
 #################
 #  Directories  #
