diff --git a/home/.bash_profile b/home/.bash_profile
index 5f54fe9..f712f92 100644
--- a/home/.bash_profile
+++ b/home/.bash_profile
@@ -8,7 +8,7 @@ export GOPATH=~/Documents/src/go
 export ANDROID_HOME=~/.local/opt/android-sdk
 
 # whether to make use of powerline fonts
-export USE_POWERLINE=0
+export USE_POWERLINE=1
 [[ -z "$DISPLAY$WAYLAND_DISPLAY$SSH_CONNECTION" ]] && USE_POWERLINE=0
 
 [[ -f ~/.bashrc ]] && . ~/.bashrc
diff --git a/home/.zshenv b/home/.zshenv
index 3fe8258..034b283 100644
--- a/home/.zshenv
+++ b/home/.zshenv
@@ -8,5 +8,5 @@ export GOPATH=~/Documents/src/go
 export ANDROID_HOME=~/.local/opt/android-sdk
 
 # whether to make use of powerline fonts
-export USE_POWERLINE=0
+export USE_POWERLINE=1
 [[ -z "$DISPLAY$WAYLAND_DISPLAY$SSH_CONNECTION" ]] && USE_POWERLINE=0
