diff --git a/.tmux.conf b/.tmux.conf
index ce0b098..81c69d4 100644
--- a/.tmux.conf
+++ b/.tmux.conf
@@ -1,4 +1,4 @@
-# for tmux 3.0
+# for tmux 2.8
 
 #############
 #  general  #
@@ -48,10 +48,14 @@ set -g window-status-format " #I ${tmux_lsub} ${tmux_window}"
 set -g window-status-current-format "${tmux_sbegin}#I ${tmux_lsub} ${tmux_window}${tmux_send}"
 set -ga update-environment "USE_POWERLINE"
 
-set-hook -g "window-layout-changed" {
-  if -b "[ '#{window_panes}' -gt 1 ]" \
-    "set pane-border-status top" "set pane-border-status off"
-}
+set-hook -g 'after-new-window' \
+  'if -b "[ \#{window_panes} -eq 1 ]" "set pane-border-status off"'
+set-hook -g 'after-kill-pane' \
+  'if -b "[ \#{window_panes} -eq 1 ]" "set pane-border-status off"'
+set-hook -g 'pane-exited' \
+  'if -b "[ \#{window_panes} -eq 1 ]" "set pane-border-status off"'
+set-hook -g 'after-split-window' \
+  'if -b "[ \#{window_panes} -gt 1 ]" "set pane-border-status top"'
 
 setenv -gu tmux_color
 setenv -gu tmux_style
@@ -87,7 +91,7 @@ set -g status-keys vi
 set -g escape-time 0
 
 # mouse
-bind -n MouseDrag1Status swap-window -d -t=
+bind -n MouseDrag1Status swap-window -t=
 bind -n WheelUpPane if -Ft= "#{||:#{mouse_any_flag},#{pane_in_mode}}" \
   "select-pane -t=; send -M" \
   "if -Ft= '#{alternate_on}' 'select-pane -t=; send -t= Up' 'select-pane -t=; copy-mode -e; send -M'"
@@ -228,7 +232,7 @@ bind -T Z Any switch-client -T prefix
 #################
 
 bind -T buffer p choose-buffer -Z
-bind -T buffer + choose-buffer -Z { run "tmux show-buffer -b '%%' | xclip -i -sel clip" }
+bind -T buffer + choose-buffer -Z 'run "tmux show-buffer -b \'%%\' | xclip -i -sel clip"'
 bind -T buffer Any switch-client -T prefix
 
 ###################
