diff --git a/home/.tmux.conf b/home/.tmux.conf
index 4d44529..194b98a 100644
--- a/home/.tmux.conf
+++ b/home/.tmux.conf
@@ -1,4 +1,4 @@
-# for tmux 2.8
+# for tmux 3.0
 
 #############
 #  general  #
@@ -41,14 +41,10 @@ set -g window-status-format " #I ${_sep_lsub} #W#{?window_flags,#{window_flags},
 set -g window-status-current-format "${_sty_sep2}${_sep_lmain1}${_sty_stat} #I ${_sep_lsub} #W#{?window_flags,#{window_flags},} ${_sty_sep1}${_sep_lmain1}"
 set -ga update-environment "USE_POWERLINE"
 
-set-hook -g 'after-new-window' \
-  'if -b "[ \#{window_panes} -eq 1 ]" "set pane-border-status off"'
-set-hook -g 'after-kill-pane' \
-  'if -b "[ \#{window_panes} -eq 1 ]" "set pane-border-status off"'
-set-hook -g 'pane-exited' \
-  'if -b "[ \#{window_panes} -eq 1 ]" "set pane-border-status off"'
-set-hook -g 'after-split-window' \
-  'if -b "[ \#{window_panes} -gt 1 ]" "set pane-border-status top"'
+set-hook -g "window-layout-changed" {
+  if -b "[ '#{window_panes}' -gt 1 ]" \
+    "set pane-border-status top" "set pane-border-status off"
+}
 
 setenv -gu _curr_mode
 setenv -gu _sty_mode
@@ -82,7 +78,7 @@ set -g status-keys vi
 set -g escape-time 0
 
 # mouse
-bind -n MouseDrag1Status swap-window -t=
+bind -n MouseDrag1Status swap-window -d -t=
 bind -n WheelUpPane if -Ft= "#{||:#{mouse_any_flag},#{pane_in_mode}}" \
   "select-pane -t=; send -M" \
   "if -Ft= '#{alternate_on}' 'select-pane -t=; send -t= Up' 'select-pane -t=; copy-mode -e; send -M'"
@@ -225,7 +221,7 @@ bind -T Z Any switch-client -T prefix
 #################
 
 bind -T buffer p choose-buffer -Z
-bind -T buffer + choose-buffer -Z 'run "tmux show-buffer -b \'%%\' | xclip -i -sel clip"'
+bind -T buffer + choose-buffer -Z { run "tmux show-buffer -b '%%' | xclip -i -sel clip" }
 bind -T buffer Any switch-client -T prefix
 
 ###################
