diff --git a/home/.bashrc b/home/.bashrc
index f3befff..9ea1842 100644
--- a/home/.bashrc
+++ b/home/.bashrc
@@ -8,7 +8,7 @@
 #
 # https://www.gnu.org/software/bash/manual/bash.html#Invoked-by-remote-shell-daemon
 
-export EDITOR="vim"
+export EDITOR="nvim"
 export LANG="en_US.UTF-8"
 export LESS="iFMR"
 export PAGER="less"
diff --git a/home/.zshenv b/home/.zshenv
index bbf412a..03b1e89 100644
--- a/home/.zshenv
+++ b/home/.zshenv
@@ -1,4 +1,4 @@
-export EDITOR="vim"
+export EDITOR="nvim"
 export LANG="en_US.UTF-8"
 export LESS="iFMR"
 export PAGER="less"
