diff --git a/home/.zshrc b/home/.zshrc
index d005ae3..3c56345 100644
--- a/home/.zshrc
+++ b/home/.zshrc
@@ -166,6 +166,8 @@ setopt no_clobber
 setopt no_flowcontrol
 autoload -Uz select-word-style && select-word-style bash
 autoload -Uz url-quote-magic && zle -N self-insert url-quote-magic
+autoload -Uz bracketed-paste-url-magic && \
+  zle -N bracketed-paste bracketed-paste-url-magic
 autoload -Uz zrecompile && zrecompile -p -R ~/.zshrc -- -M ~/.zcompdump
 
 # Tell Apple Terminal the working directory
